src/libraries/System.Linq.Expressions/src/System/Dynamic/DynamicObject.cs

```diff
@@ -47,7 +48,7 @@ protected DynamicObject()
         /// <param name="result">The result of the get operation.</param>
         /// <returns>true if the operation is complete, false if the call site should determine behavior.</returns>
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1007:UseGenericsWhereAppropriate")]
-        public virtual bool TryGetMember(GetMemberBinder binder, out object result)
+        public virtual bool TryGetMember(GetMemberBinder binder, [NotNullWhen(true)] out object? result)
```

Comment:
>[NotNullWhen(true)] [](start = 65, length = 19)

Can't an implementation return `null` for the value of a property?

Reply:
You mean result can be null even true returned? Seems not all implementation guarantee non null result on true return ...

Reply:
Yes, the member being invoked may be a property that returns a nullable value, say `object? Property => null;`. Given that, the `[NotNullWhen(true)]` attribute should be removed.

---
In reply to: [362647335](https://github.com/dotnet/runtime/pull/1064#discussion_r362647335) [](ancestors = 362647335)